code/math/vecmat.cpp

```diff
@@ -2571,6 +2571,56 @@ void vm_vec_boxscale(vec2d *vec, float  /*scale*/)
 	vec->y *= ratio;
 }
 
+// adds one matrix to another.
+// dest can equal source
+// dest += src
+void vm_matrix_add2(matrix* dest, const matrix* src)
+{
+	dest->vec.fvec += src->vec.fvec;
+	dest->vec.rvec += src->vec.rvec;
+	dest->vec.uvec += src->vec.uvec;
+}
```

Comment:
I would strongly advise making a separate PR for *just* the vecmat changes, and then rebasing this PR after that one is in the codebase.  The reason is I think it would be good to include a few more things here...

* `vm_matrix_add` for two source arguments, and a `vm_matrix_sub`
* `vm_matrix_sub2` equivalent to `vm_matrix_add2`
* some test suite tests to verify that the matrix math is, in fact, correct